osu.Game/Overlays/Comments/CommentsContainer.cs

```diff
@@ -121,12 +119,24 @@ private void load()
 
         protected override void LoadComplete()
         {
-            Sort.BindValueChanged(onSortChanged, true);
+            Sort.BindValueChanged(_ => resetComments());
             base.LoadComplete();
         }
 
-        private void onSortChanged(ValueChangedEvent<CommentsSortCriteria> sort)
+        /// <param name="type">The type of resource to get comments for.</param>
+        /// <param name="id">The id of the resource to get comments for.</param>
+        public void ShowComments(CommentableType type, long id)
         {
+            this.type = type;
+            this.id = id;
+            Sort.TriggerChange();
+        }
+
+        private void resetComments()
+        {
+            if (id == default)
```

Comment:
- using `default` for primitive type (long) looks bad
- will anywhere accept 0 for valid id? should it be `long?` ?
- if it's reset to show nothing, clear should be invoked

Reply:
Looks like reset is a bad name. Should be `refetch` instead (?)
Yeah, nullable long is a better thing to use

Reply:
Just a side note: we will never use this method manually in-game: all the id's will be used from overlays themselves, so it's not really about allowing or disallowing to use 0 for valid id. That's why I changed my code to use `long?` instead of checking `id==0`. But will see, hot it goes.